Gordon360/Services/DirectMessageService.cs

```diff
@@ -290,30 +328,35 @@ public bool StoreUserConnectionIds(String userId, String connectionId)
 
         }
 
-        public IEnumerable<ConnectionIdViewModel> GetUserConnectionIds(String userId)
+        public List<IEnumerable<ConnectionIdViewModel>> GetUserConnectionIds(List<String> userIds)
         {
             var _unitOfWork = new UnitOfWork();
-            var query = _unitOfWork.AccountRepository.FirstOrDefault(x => x.gordon_id == userId);
-            if (query == null)
-            {
-                throw new ResourceNotFoundException() { ExceptionMessage = "The account was not found." };
+            foreach(string user in userIds) {
+                var query = _unitOfWork.AccountRepository.FirstOrDefault(x => x.gordon_id == user);
+                if (query == null)
+                {
+                    throw new ResourceNotFoundException() { ExceptionMessage = "One of the accounts was not found." };
+                }
             }
+            var idList = new List<IEnumerable<ConnectionIdViewModel>>(300);
 
-            var userIdParam = new SqlParameter("@user_id", userId);
-
-            var result = RawSqlQuery<ConnectionIdViewModel>.query("GET_ALL_CONNECTION_IDS_BY_ID @user_id", userIdParam); //run stored procedure
-
-            var model = result.Select(x =>
+            foreach (string user in userIds)
             {
-                ConnectionIdViewModel y = new ConnectionIdViewModel();
-                y.connection_id = x.connection_id;
-               
+                var userIdParam = new SqlParameter("@user_id", user);
 
-                return y;
-            });
+                var result = RawSqlQuery<ConnectionIdViewModel>.query("GET_ALL_CONNECTION_IDS_BY_ID @user_id", userIdParam); //run stored procedure
 
+                var model = result.Select(x =>
+                {
+                    ConnectionIdViewModel y = new ConnectionIdViewModel();
+                    y.connection_id = x.connection_id;
+                    return y;
+                });
```

Comment:
I thought that was necessary for the mapping to the correct variables from the result in the DB?

Reply:
In this case it definitely isn't necessary. Since you're specifying in the query that the return type is ConnectionIdViewModel, the data is already being cast into an enumerable of ConnectionIdViewModel objects. You can verify this by hovering over `result` in Visual Studio and the intellisense will show you that it's an `IEnumerable<ConnectionIdViewModel>`. Also, notice that you're setting the connection_id of `y` equal to x's `connection_id`. That is only possible if x already has the correct data in the correct ConnectionIdViewModel format, since `connection_id` is a property of objects of type `ConnectionIdViewModel`. Since x is a variable representing the elements in the `result` `IEnumerable`, x is a `ConnectionIdViewModel` object.

The use case for `Select` is when you have data in one format and want to convert it into another. The data you're getting from the backend is already being returned in the correct format so it isn't necessary here.